extensions/amp-script/0.1/amp-script.js

```diff
@@ -288,24 +288,30 @@ export class AmpScript extends AMP.BaseElement {
     };
 
     // Create worker and hydrate.
-    WorkerDOM.upgrade(
-      container || this.element,
-      workerAndAuthorScripts,
-      config
-    ).then((workerDom) => {
-      this.workerDom_ = workerDom;
-      this.initialize_.resolve();
-      // workerDom will be null if it failed to init.
-      if (this.workerDom_) {
-        this.workerDom_.onerror = (errorEvent) => {
-          errorEvent.preventDefault();
-          user().error(
-            TAG,
-            `${errorEvent.message}\n    at (${errorEvent.filename}:${errorEvent.lineno})`
-          );
-        };
-      }
-    });
+    WorkerDOM.upgrade(container || this.element, workerAndAuthorScripts, config)
+      .then((workerDom) => {
+        this.workerDom_ = workerDom;
+        this.initialize_.resolve();
+        // workerDom will be null if it failed to init.
+        if (this.workerDom_) {
+          this.workerDom_.onerror = (errorEvent) => {
+            errorEvent.preventDefault();
+            user().error(
+              TAG,
+              `${errorEvent.message}\n    at (${errorEvent.filename}:${errorEvent.lineno})`
+            );
+          };
+        }
+      })
+      .catch((err) => {
+        // Catch errors if due to an issue with workerAndAuthorScripts rejection.
```

Comment:
Can we just chain a `catch()` onto `workerAndAuthorScripts` above?

Reply:
`WorkerDOM.upgrade` returns a promise which is the uncaught one. The more obvious thing to do IMO would be to return that promise instead of `workerAndAuthorScripts` down below. Is there any reason we didn't do that?

Reply:
Oh let's just do that. Don't think so.